packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx

```diff
@@ -61,10 +61,12 @@ export interface BaseNonStaticPickerProps {
 export interface BaseSingleInputNonStaticPickerProps {
   /**
    * The label content.
+   * Will be ignored on multi input range pickers.
```

Comment:
Maybe we could at least duplicate these types in `-pro` package to avoid bleeding this redundant information to generic (non range) pickers? 🤔 
Maybe it would also be possible to have live typing depending on props, so that we could only show this kind of doc when `!isSingleInput`... But if we'd do the first step, then IMHO this would become less important. 🤷

Reply:
For the dynamic JSDoc depending on the prop, it's maybe doable but I don't think the complexity is worth it